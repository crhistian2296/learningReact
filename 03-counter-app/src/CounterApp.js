import React, { useState } from "react";
import PropTypes from "prop-types";

//las props entran como objetos que usualmente se desestructuran en los parametros del componente
const CounterApp = ({ value }) => {
    const [counter, setcounter] = useState(value);

    //handleAdd
    const handleAdd = () => {
        //setcounter(counter + 1);
        setcounter((a) => a + 1);
    };
    const handleSubtract = () => {
        //setcounter(counter + 1);
        setcounter((a) => a - 1);
    };
    const handleReset = () => {
        //setcounter(counter + 1);
        setcounter(() => value);
    };

    return (
        <>
            <h1>CounterApp</h1>
            <h2 id="contador"> {counter} </h2>

            <button id="bAdd" onClick={handleAdd}>
                +1
            </button>
            <button id="bReset" onClick={handleReset}>
                Reset
            </button>
            <button id="b-1" onClick={handleSubtract}>
                -1
            </button>
        </>
    );
};

CounterApp.prototype = {
    value: PropTypes.number.isRequired,
};

CounterApp.defaultProps = {
    value: 10,
};

export default CounterApp;
