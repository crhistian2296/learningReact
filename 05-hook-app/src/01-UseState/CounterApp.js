import React from "react";
// import "./Counter.css";
import useCounter from "./hooks/useCounter";

const CounterApp = () => {
  const { state, increment, decrement, reset } = useCounter(400);

  return (
    <div className="m-5">
      <h2>Counter with hook = {state}</h2>
      <hr />
      <button className="btn btn-primary m-1" onClick={() => increment()}>
        +1
      </button>
      <button className="btn btn-primary m-1" onClick={() => decrement(2)}>
        -2
      </button>
      <button className="btn btn-info text-light m-1" onClick={() => reset()}>
        reset
      </button>
    </div>
  );
};

export default CounterApp;
