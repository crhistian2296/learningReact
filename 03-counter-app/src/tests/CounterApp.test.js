import React from "react";
import { shallow } from "enzyme";
import CounterApp from "../CounterApp";

describe("Pruebas en CounterApp", () => {
    //Funciones
    const setCounterApp = (valor) => (valor ? shallow(<CounterApp value={valor} />) : shallow(<CounterApp />));
    const getCounterValue = (wrapper) => parseInt(wrapper.find("#contador").text().trim());

    //Iniciales
    let wrapper = setCounterApp(); //Mala practica pero no pierdo el intelisense
    beforeEach(() => {
        //Ejecuta antes de evaluar cada test
        wrapper = setCounterApp(); //reinicia el componente a sus calores por defecto
    });

    test("Debe de mostrar <CounterApp /> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("Debe hacer match en 100", () => {
        const wrapper = setCounterApp(100);
        const contador = getCounterValue(wrapper);

        expect(contador).toBe(100);
    });

    test("Debe de incrementar con el boton +1", () => {
        //Se simula el click
        wrapper.find("#b1").simulate("click");
        //Se obtiene el valor nuevo del contador
        const contadorNuevo = getCounterValue(wrapper);
        //console.log(contadorNuevo);
        expect(contadorNuevo).toBe(11);
    });

    test("Debe de decrementar con el boton -1 ", () => {
        wrapper.find("#b3").simulate("click");
        const contadorNuevo = getCounterValue(wrapper);
        //console.log(contadorNuevo);
        expect(contadorNuevo).toBe(9);
    });
});
