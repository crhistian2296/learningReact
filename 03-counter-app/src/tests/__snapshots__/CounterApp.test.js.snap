// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en CounterApp Debe de mostrar <CounterApp /> correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2
    id="contador"
  >
     
    10
     
  </h2>
  <button
    id="b1"
    onClick={[Function]}
  >
    +1
  </button>
  <button
    id="b2"
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    id="b3"
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
